core/backend/src/workspace/SettingsSchemas.ts

```diff
@@ -21,28 +21,32 @@ import { IModelJsFs } from "../IModelJsFs";
  * @beta
  */
 export interface SettingSchema extends Readonly<JSONSchema> {
-  readonly items?: SettingSchema; // must be single object, not array
-  readonly type: JSONSchemaTypeName; // type is required for settings
+  /** entries for an array. Must be single object, not array */
+  readonly items?: SettingSchema;
+  /** type is required for settings */
+  readonly type: JSONSchemaTypeName;
+  /** name of typeDef for objects or arrays to inherit */
+  readonly extends?: string;
+  /** for objects, a list of named properties and their definitions */
   readonly properties?: { [name: string]: SettingSchema };
-
   /** whether the setting replaces lower priority entries with the same name or combines with them. */
   readonly cumulative?: true;
```

Comment:
Looks like this is the property that's supposed to control whether querying an array property returns the highest-priority match or a concatenation of all the unique values in all matching settings. Presumably it is meaningless for non-arrays (including objects, though you could make a case for a higher-priority setting overriding some properties of a lower-priority one).

This is not used anywhere currently.

Uniqueness is straightforward for things that aren't objects or arrays (assuming strings are compared case-sensitively).

Reply:
I added this for Settings editors, so they can indicate whether the values supplied are meant to override or combine with other values. It is only useful for arrays, but there's no way to have a schema member that only applies to arrays.

Reply:
Expectation is that `getArray` will look up the setting def, check this flag, and combine or override as specified, no? That's the missing piece.

Reply:
That sounds like a good idea. I guess the default should be `false` if there's no schema?

Reply:
Yes. Implemented it.